### ISSUE TITLE
fix(inference): n-gram speculative decoding is lossy — verify_draft commits draft[0] unverified (diverges from greedy)

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#243]
## Bug

`verify_draft` (crates/inference/src/speculative.rs:1835) commits the **first** draft token without verifying it against the target model's prediction at the current position. This makes n-gram speculative decoding **lossy** — its output can diverge from plain greedy decoding, when speculative decoding is supposed to be exact (token-identical to greedy).

```rust
// speculative.rs ~1846
for (i, &draft_token) in draft_tokens.iter().enumerate() {
    let logits = forward_fn(draft_token, position_start + i);
    let model_choice = argmax(&logits);
    all_logits.push(logits);
    accepted += 1;                                   // draft[i] counted accepted unconditionally
    if i + 1 < draft_tokens.len() && model_choice != draft_tokens[i + 1] as usize {
        break;                                       // only ever rejects draft[i+1], never draft[0]
    }
}
```

`forward_fn(draft[0], pos)` conditions on `draft[0]` already being the token at `pos` and returns the logits for `pos+1`. The target's actual greedy choice **at `pos`** (i.e. `argmax(forward_fn(all_tokens.last(), pos))`) is never computed, so `draft[0]` is never checked. A length-1 draft is therefore always accepted verbatim.

## Trigger

`prompt_tokens = [1,1,1]` (n-gram speculator drafts `[1]`), `max_new_tokens = 1`, `forward_fn` whose greedy choice at the current position is `0` (not `1`): the function returns `[1]`, but greedy decoding would return `[0]`.

## Impact

Speculative decoding silently changes generation output vs the non-speculative path. Any parity/correctness expectation that "speculation is a pure speedup" is violated. Confidence: medium-high (requires the n-gram path to draft a token greedy wouldn't pick — realistic on repetitive context).

## Fix (needs design)

Restructure verification so the current-position target distribution gates `draft[0]`:
- Either run a forward pass for the current position first and verify `draft[0] == argmax(current_logits)` before consuming it, or
- Reframe the speculator so `draft[0]` is the already-verified bonus token from the prior step (standard speculative-decoding bookkeeping).

This is an algorithmic change to the verification loop, not a one-line clamp — filing separately from the budget-overrun bug. Discovered via a sweep, TBV-confirmed against the source.
